### PR TITLE
Fix privileges for predefined roles

### PR DESCRIPTION
**Pull request by jkakavas**
In #71904 we added two new predefined roles, editor and viewer in
elasticsearch. This change adds a missing application privilege to
these roles. This is only needed in 7.x as the behavior has changed
in Kibana, starting 8.0 and this extra privilege is not necessary.

Marking as non-issue as this is not yet released.

### CODE REVIEW

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java [hunk 1]
```diff
@@ -356,7 +356,11 @@ private static RoleDescriptor buildViewerRoleDescriptor() {
                 RoleDescriptor.ApplicationResourcePrivileges.builder()
                     .application("kibana-.kibana")
                     .resources("*")
-                    .privileges("read").build() },
+                    .privileges("read").build(),
+                RoleDescriptor.ApplicationResourcePrivileges.builder()
+                    .application("kibana-.*")
```
Comment (bytebilly) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java:361]:
```suggestion
                    .application("kibana-*")
```

I think this is the right syntax, given the comment in https://github.com/elastic/elasticsearch/pull/71710#discussion_r614007999 and a chat I had with @legrego 

I'll leave to Larry to comment futher.

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java [hunk 2]
```diff
@@ -383,7 +387,11 @@ private static RoleDescriptor buildEditorRoleDescriptor() {
                 RoleDescriptor.ApplicationResourcePrivileges.builder()
                     .application("kibana-.kibana")
                     .resources("*")
-                    .privileges("all").build() },
+                    .privileges("all").build(),
+                RoleDescriptor.ApplicationResourcePrivileges.builder()
+                    .application("kibana-.*")
```
Comment (bytebilly) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java:392]:
```suggestion
                    .application("kibana-*")
```

```suggestion
                    .application("kibana-*")
```

I think this is the right syntax, given the comment in https://github.com/elastic/elasticsearch/pull/71710#discussion_r614007999 and a chat I had with @legrego 

I'll leave to Larry to comment futher.

**Review by bytebilly — CHANGES_REQUESTED** [review]
I left a couple of changes that I think are necessary to make it work properly.

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java (hunk 1 above)
Comment (jkakavas) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java:361]: You're probably right, let's wait for @legrego to verify

Comment (legrego) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java:361]: @jkakavas, @bytebilly's suggestion is the correct syntax: There should not be a `.` in the application name for "reserved" privileges

**Comment by jkakavas** [comment]
@elasticmachine update branch

**Review by bytebilly — APPROVED** [review]
LGTM 👍

**Pull request merged**